pkg/gadgets/socket-collector/tracer/tracer.go

```
@@ -0,0 +1,268 @@
+}
+
+func prifixedLog(s string) {
+	log.Printf("%s: %s", label, s)
+}
+
+func parseIP(ipHex string) string {
+	ipBytes, err := hex.DecodeString(ipHex)
+	if err != nil {
+		prifixedLog(fmt.Sprintf("Failed to decode hex-string IP %q: %s", ipHex, err))
+		return ""
+	}
+
+	ip := net.IP(ipBytes)
+	return ip.String()
+}
+
+func parsePort(portHex string) uint16 {
+	portBytes, err := hex.DecodeString(portHex)
+	if err != nil {
+		prifixedLog(fmt.Sprintf("Failed to decode hex-string Port %q: %s", portHex, err))
+		return 0
+	}
+
+	return binary.BigEndian.Uint16(portBytes)
+}
+
+func parseTCPStatus(statusHex string) string {
+	statusBytes, err := hex.DecodeString(statusHex)
+	if err != nil {
+		prifixedLog(fmt.Sprintf("Failed to decode hex-string TCP status %q: %s", statusHex, err))
+		return ""
+	}
+
+	switch uint8(statusBytes[0]) {
+	case TCP_ESTABLISHED:
+		return "ESTABLISHED"
+	case TCP_SYN_SENT:
+		return "SYN_SENT"
+	case TCP_SYN_RECV:
+		return "SYN_RECV"
+	case TCP_FIN_WAIT1:
+		return "FIN_WAIT1"
+	case TCP_FIN_WAIT2:
+		return "FIN_WAIT2"
+	case TCP_TIME_WAIT:
+		return "TIME_WAIT"
+	case TCP_CLOSE:
+		return "CLOSE"
+	case TCP_CLOSE_WAIT:
+		return "CLOSE_WAIT"
+	case TCP_LAST_ACK:
+		return "LAST_ACK"
+	case TCP_LISTEN:
+		return "LISTEN"
+	case TCP_CLOSING:
+		return "CLOSING"
+	case TCP_NEW_SYN_RECV:
+		return "NEW_SYN_RECV"
+	default:
+		prifixedLog(fmt.Sprintf("Invalid TCP status: %s (%v)", statusHex, statusBytes))
+	}
+
+	return "UNKNOWN"
+}
+
+func parseUDPStatus(statusHex string) string {
+	statusBytes, err := hex.DecodeString(statusHex)
+	if err != nil {
+		prifixedLog(fmt.Sprintf("Failed to decode hex-string UDP status %q: %s", statusHex, err))
+		return ""
+	}
+
+	switch uint8(statusBytes[0]) {
+	case TCP_ESTABLISHED:
+		return "ACTIVE"
+	case TCP_CLOSE:
+		return "INACTIVE"
+	default:
+		prifixedLog(fmt.Sprintf("Unexpected UDP status: %s (%v)", statusHex, statusBytes))
+	}
+
+	return "UNKNOWN"
+}
+
+func getTCPIter() *link.Iter {
```

perhaps this should return the error instead of printing it... :thinking:, then the caller should care about printing it.

---

Yeah, I think it should return the error and that information should be passed back to `operationError`, otherwise if something goes wrong the user can't easily know what it was.

---

Yeah, I also discussed this today with @marga-kinvolk, I will implement this change.